### PR TITLE
Fixes #17441 - override all parameters permission changed to edit

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

File: test/controllers/puppetclasses_controller_test.rb [hunk 1]
```diff
@@ -198,4 +202,23 @@ def test_override_none
     assert_match /No parameters to override/, flash[:error]
     assert_redirected_to puppetclasses_url
   end
+
+  test 'user with edit_pupetclasses permission should succeed in overriding all parameters' do
```
Comment (domcleal) [inline test/controllers/puppetclasses_controller_test.rb]: spelling: edit_puppetclasses

File: test/controllers/puppetclasses_controller_test.rb [hunk 2]
```diff
@@ -198,4 +202,23 @@ def test_override_none
...
+    setup_user "edit", "puppetclasses"
+    env = FactoryGirl.create(:environment)
+    pc = FactoryGirl.create(:puppetclass, :with_parameters, :environments => [env])
+    refute pc.class_params.first.override
+    post :override, {:id => pc.to_param, :enable => 'true'}, set_session_user.merge(:user => users(:one).id)
+    assert_match /overridden all parameters/, flash[:notice]
+    assert_redirected_to puppetclasses_url
+  end
+
+  test 'user without edit_pupetclasses permission should fail in overriding all parameters' do
```
Comment (domcleal) [inline test/controllers/puppetclasses_controller_test.rb]: ditto, spelling

**Review by domcleal — CHANGES_REQUESTED** [review]

**Comment by orrabin** [comment]
@domcleal thanks, I fixed the spelling and rebased.

**Review by domcleal — APPROVED** [review]

**Comment by domcleal** [comment]
Merged, thanks @orrabin.

**Pull request merged**